pin/gc/gc.go

```diff
@@ -17,18 +18,19 @@ var log = logging.Logger("gc")
 // GC performs a mark and sweep garbage collection of the blocks in the blockstore
 // first, it creates a 'marked' set and adds to it the following:
 // - all recursively pinned blocks, plus all of their descendants (recursively)
+// - the root directory for the files API, plus all of its descendants (recursively)
 // - all directly pinned blocks
 // - all blocks utilized internally by the pinner
 //
 // The routine then iterates over every block in the blockstore and
 // deletes any block that is not found in the marked set.
-func GC(ctx context.Context, bs bstore.GCBlockstore, pn pin.Pinner) (<-chan key.Key, error) {
+func GC(ctx context.Context, bs bstore.GCBlockstore, pn pin.Pinner, filesRoot *mfs.Root) (<-chan key.Key, error) {
```

Comment:
@whyrusleeping in commit ed601f6 I passed in the dag Node inside of the files root, this eliminated the extra dependency.  In the second commit 174d2ce I passed in an array of keys to use as the root set for the best effort pins; this is a bit more general but involves a bit more code.  Let me know what version you like best.